contracts/components/Safes.sol

```diff
@@ -257,4 +285,22 @@ contract Safes {
 
         return true;
     }
+
+    /**
+     * @notice Update the D-Day
+     * @param _safeId Id of the safe
+     * @param _DDay The timestamp after which the beneficiary can directly claim the safe
+     */
+    function _updateDDay(string memory _safeId, uint256 _DDay)
+        internal
+        DDayUpdate(_safeId)
+        returns (bool)
+    {
+        Types.Safe memory safe = safes[_safeId];
+
+        safe.dDay = block.timestamp + _DDay;
```

Comment:
Unlike signalBased, the DDay timestamp is an absolute time. _DDay itself should be a future timestamp.